regression/run/typemap/main.f

```diff
@@ -0,0 +1,64 @@
+! Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+! other Shroud Project Developers.
+! See the top-level COPYRIGHT file for details.
+!
+! SPDX-License-Identifier: (BSD-3-Clause)
+! #######################################################################
+!
+! Test Fortran API generated from typemap.yaml.
+!
+program tester
+  use fruit
+  use iso_c_binding
+  use typemap_mod
+  implicit none
+  logical ok
+
+  call init_fruit
+
+  call test_indextype
+
+  call fruit_summary
+  call fruit_finalize
+
+  call is_all_successful(ok)
+  if (.not. ok) then
+     call exit(1)
+  endif
+
+contains
+
+  subroutine test_indextype
+    integer(INDEXTYPE) :: indx
+    integer(C_INT32_T) :: indx32
+    integer(C_INT64_T) :: indx64
+
+    indx = 0
+!    call pass_index(indx)
+
+    ! Match files with C.
+    indx32 = 2
+    indx64 = 2_C_INT64_T**34
+#if defined(USE_64BIT_INDEXTYPE)
```

Comment:
Minor: does the file extension for this file need to be capitalized (for some compilers) since it needs to be preprocessed (e.g. `.F`  instead of `.f`) ?

I'm asking more with regards to the generated files and implications for user codes than for this specific example.

Reply:
[I tried responding before but I think it got lost]
When compiling Fortran files as part of the test, I always add the flag to use the C preprocessor.  I find that simpler than dealing with two different file suffixes.  I do that with other Fortran projects too.